021_MergeTwoSortedLists.py

```diff
@@ -0,0 +1,84 @@
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+
+class Solution(object):
+    def mergeTwoLists(self, l1, l2):
+        last=None
+        lr=None
+        while (l1!=None) and (l2!=None):
+            if l1.val<=l2.val:
+                if lr==None:
+                    lr=l1
+                    last=l1
+                else: 
+                    last.next=l1
+                    last=last.next
```

Comment:
line 18 and line 21 are doing the same thing, "last = l1". You can remove that from if/else and add it outside before the "l1 = l1.next"

Reply:
Same for the 2nd part, when you are looking into l2